### PR TITLE
Backport: Clean up & improve PK write test functions

### PR DESCRIPTION
**Pull request by valeriosetti**
This is the backport of issue #7446 as solved in PR #7449.

## Gatekeeper checklist

- [x] **changelog** not required
- [x] **backport** not required: this is the backport
- [x] **tests** not required

### CODE REVIEW

**Review by mpg — APPROVED** [review]
LGTM

**Comment by mpg** [comment]
Note for the future: as a reviewer, I like it when backports have the same commit structure as the original PR, because then I can use `git range-diff` to check it's just the same changes. (Also, as a backport author, that's easily achieved by using `git cherry-pick development..original_branch`.) (Here I manually made a diff of diffs, but I don't know how to make it as readable as the output from git range-diff.)

**Review by AndrzejKurek — APPROVED** [review]

File: tests/data_files/Makefile [hunk 1]
```diff
@@ -888,6 +888,57 @@ ec_prv.pk8param.pem: ec_prv.pk8param.der
 	$(OPENSSL) pkey -in $< -inform DER -out $@
 all_final += ec_prv.pk8param.pem
 
+################################################################
+#### Convert PEM keys in DER format
```
Comment (AndrzejKurek) [inline tests/data_files/Makefile:892]:
```suggestion
#### Convert PEM keys to DER format
```
Same suggestion as in the original PR.

**Review by mpg — CHANGES_REQUESTED** [review]
This needs updating to match changes in the original PR.

**Review by mpg — APPROVED** [review]
LGTM, thanks!

**Review by AndrzejKurek — APPROVED** [review]
Faithful backport.

**Pull request merged**